src/pybind/ceph_volume_client.py

```diff
@@ -1442,6 +1451,34 @@ def get_object(self, pool_name, object_name):
             ioctx.close()
         return bytes_read
 
+    def update_object(self, pool_name, object_name, data_modifier):
+        iterations = 0
+
+        while True:
+            data, version1 = self.get_object_and_version(pool_name, object_name)
+            result_data = data_modifier(data)
+            version2 = self.put_object(pool_name, object_name, result_data)
```

Comment:
The put should fail if the version is out-of-date. Otherwise this is not atomic.

Reply:
Okay. There should be no iterarions i.e. `update_object()` should simply return if `put_object()` fails, right?

Reply:
@batrick I am getting rid of iterations. Let me know if you think otherwise.